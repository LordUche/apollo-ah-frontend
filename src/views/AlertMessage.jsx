import React from 'react';
import PropTypes from 'prop-types';

/**
 *
 *
 * @param {*} {props} - Properties passed in from the parent component
 * @returns {JSX} - The JSX that will represent the alert message on the UI
 */
function AlertMessage(props) {
  const { resetState, status, notification } = props;
  return (
    <div>
      <p
        className={
          `alert ${(status === 'success') ? 'alert-success' : 'alert-danger'} alert-dismissible fade show`
        }
        role="alert"
      >
        {notification}
        <button
          type="button"
          className="close"
          aria-label="Close"
          onClick={resetState}
        >
          <span aria-hidden="true">&times;</span>
        </button>
      </p>
    </div>
  );
}

AlertMessage.propTypes = {
  notification: PropTypes.string,
  resetState: PropTypes.func,
  status: PropTypes.string,
};

AlertMessage.defaultProps = {
  notification: '',
  resetState: f => f,
  status: '',
};

export default AlertMessage;
