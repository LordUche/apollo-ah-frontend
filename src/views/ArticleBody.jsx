import React from 'react';
import PropTypes from 'prop-types';
import Tags from './ArticleTags';
import clapImage from '../images/clap.svg';
/**
 *
 * @param {*} props - The props passed from the parent Article component
 * @returns {JSX} - The JSX representation of the Article's bpdy
 */
function ArticleBody(props) {
  const { bookmarkArticle, bookmarked, article } = props;
  return (
    <div>
      <h1 className="signle-title">{article.title}</h1>
      <p
        className="single-sub"
        style={{ fontSize: 21, color: '#9c9c9c', fontWeight: 'bold' }}
      >
        {article.description}
        {' '}
      </p>
      <div className="single__author__container">
        <div className="article__author">
          <div className="author__profile--mini">
            <img src={article.image} alt="" />
            <div className="author__profile--details">
              <h6>Alexandra</h6>
              <span>22 hrs ago</span>
              <span className="article-author-dot" />
              <span>4 min read</span>
            </div>
          </div>
          <div className="bookmark__div">
            <button onClick={bookmarkArticle} type="button">
              <span className={`fas fa-bookmark transition ${(bookmarked) ? 'bookmarked' : ''}`} />
            </button>
          </div>
        </div>
      </div>
      <div className="single-body">
        {article.body}
      </div>
      <Tags />
      <div className="pg-empty-placeholder" />
      <div className="clap-grp">
        <span className="clap-icon">
          <img src={clapImage} alt="" />
          40k claps
        </span>
        <div className="share-grp">
          <a href="you" className="text-danger"><i className="fas fa-envelope" /></a>
          <a href="you" className="text-primary"><i className="fab fa-facebook-square" /></a>
          <a href="you" className="text-white"><i className="fab fa-twitter-square text-info" /></a>
          <a href="you" className="text-muted"><i className="fas fa-bookmark" /></a>
          <a href="you" className="text-muted"><i className="fas fa-ellipsis-h" /></a>
        </div>
      </div>
      <div className="single-author-container single-article-author">
        <div className="article-author">
          <img src={article.image} alt="" />
          <div>
            <h6>Alexandra</h6>
            <p>Journalist, author, dad. Former TIME magazine editor and foreign correspondent. </p>
            <button type="button" className="btn btn-light">Follow</button>
          </div>
        </div>
      </div>
    </div>
  );
}

ArticleBody.propTypes = {
  bookmarkArticle: PropTypes.func.isRequired,
  bookmarked: PropTypes.bool.isRequired,
  article: PropTypes.object.isRequired,
};

export default ArticleBody;
