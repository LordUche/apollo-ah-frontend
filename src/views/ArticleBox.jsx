import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import compareDateFormat from 'date-fns/distance_in_words_strict';

export default function ArticleBox({ article }) {
  return (
    <div className="col-sm-6 col-lg-4">
      <div className="article">
        <Link to={`/article/${article.slug}`} className="link-surface" />
        <div className="article-cover">
          <img src={article.image} className="article-cover-img" alt={article.title} />
        </div>
        <div className="article-item">
          <span className="article-category">{article.articleCategory.category}</span>
          <h4>{article.title}</h4>
          <p>{article.description.substring(0, 100)}</p>
          <div className="article-author">
            <img src={article.User.Profile.image} alt={article.User.Profile.username} />
            <div>
              <h6>{article.User.Profile.firstname || article.User.Profile.username}</h6>
              <span>
                {compareDateFormat(article.createdAt, new Date())}
                {' '}
                 ago
              </span>
              <span className="article-author-dot" />
              <span>{article.readTime}</span>
            </div>
          </div>
        </div>
      </div>
    </div>

  );
}
ArticleBox.propTypes = {
  article: PropTypes.object.isRequired,
};
