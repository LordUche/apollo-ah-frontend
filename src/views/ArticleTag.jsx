import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
/**
 * @returns {JSX} - The JSX representation of the article's tags
 *
 */
function Tag({ tag }) {
  return (
    <Link to={`/search?tag=${tag.tagName}`} className="badge">
      {tag.tagName}
    </Link>
  );
}
Tag.propTypes = {
  tag: PropTypes.object.isRequired,
};

export default Tag;
