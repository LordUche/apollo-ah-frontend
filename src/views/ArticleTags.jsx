import React from 'react';

/**
 * @returns {JSX} - The JSx representation of the article's tags
 */
function Tags() {
  return (
    <div className="single-tags text-center">
      <span className="badge">andela</span>
      <span className="badge">technology</span>
      <span className="badge">bootcamp</span>
      <span className="badge">computer</span>
      <span className="badge">laptop</span>
    </div>
  );
}

export default Tags;
