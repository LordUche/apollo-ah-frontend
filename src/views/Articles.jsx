import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import time from '../utils/time';

/**
 * @description - returns all articles created by all authors
 * @param {props} articles - all authors article
 * @return {JSX}
 */
const Articles = ({ articles }) => (
  <div className="container">
    <div className="row">
      {articles.map(article => (
        <div key={article.id} className="col-lg-4 col-sm-6">
          <div className="article">
            <Link to="/single-article" className="link-surface" />
            <div className="article-cover">
              <img
                  src={article.image}
                  className="article-cover-img"
                  alt="article"
                />
            </div>
            <div className="article-item">
              <span className="article-category">{article.articleCategory.category}</span>
              <h4>{article.title}</h4>
              <p>
                {
                    article.description.length <= 50 ? article.description.substring(0, 50) : `${article.description.substring(0, 50)}...`
                  }
              </p>
              <div className="article-author">
                <img alt="article" src={article.User.Profile.image} />
                <div>
                  <h6>{article.User.Profile.username}</h6>
                  <span>
                    {`${time(article)} ago`}
                  </span>
                  <span className="article-author-dot" />
                  <span>{article.readTime}</span>
                  <i
                      className="fas fa-bookmark transition"
                    />
                </div>
              </div>
            </div>
          </div>
        </div>
      ))}
    </div>
  </div>
);

Articles.propTypes = {
  articles: PropTypes.array,
};

Articles.defaultProps = {
  articles: [],
};

export default Articles;
