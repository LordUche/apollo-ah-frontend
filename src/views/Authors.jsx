/* eslint-disable react/no-array-index-key */
/* eslint-disable react/require-default-props */
/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * @description - returns all author's or selected author's
 * @param {props} authors - all authors
 * @param {props} begin - start of range of author's to select
 * @param {props} end - end of range of author's to select
 * @param {props} follow - follow feature
 * @return {JSX}
 */
const Authors = ({ authors, begin, end, follow }) => {
  return (
    <div className="container">
      <div className="row">
        {
          authors.length > 0 ?
            authors.slice(begin, end).map((author, index) => {
              const { firstname, lastname } = author.authorsProfile;
              return (
                <div className="col-sm-6 col-lg-4" key={author.id}>
                  <div className="author">
                    <img alt="author" src={author.authorsProfile.image} />
                    <div data-pg-collapsed>
                      <h4>{`${firstname} ${lastname}`}</h4>
                      <p>{author.authorsProfile.bio}</p>
                      <button
                        onClick={follow}
                        className="transition btn"
                        type="button"
                      >
                        Follow
                      </button>
                    </div>
                  </div>
                </div>
              );
            }) : ''
        }
      </div>
    </div>
  </div>
);

Authors.propTypes = {
  authors: PropTypes.array,
  begin: PropTypes.number,
  end: PropTypes.number,
  follow: PropTypes.func,
};

Authors.defaultProps = {
  authors: [],
  begin: 1,
  end: 10,
  follow: f => f,
};

export default Authors;
