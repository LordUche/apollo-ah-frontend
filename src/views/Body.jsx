/* eslint-disable no-nested-ternary */
import React from 'react';
import { Link } from 'react-router-dom';
import Proptypes from 'prop-types';
import FooterImage from '../../public/images/background/Image.png';

/**
 * @description - returns the body of the landing page
 * @param {props} loadingCategory - loading status for categories
 * @param {props} PlaceholderLoader - placeholder element to show when articles is
 * been fetched from the API
 * @param {props} articlesCategory - categories belonging to articles
 * @param {props} loadingArticles - loading status for articles
 * @param {props} Articles - prop in representing articles
 * @param {props} articles - all articles
 * @param {props} Authors - all authors
 * @param {props} fiveStarAuthors - authors with highest rarting
 * @return {JSX}
 */
const Body = ({
  loadingCategory,
  PlaceholderLoader,
  Category,
  articlesCategory,
  loadingArticles,
  Articles,
  allArticles,
  Authors,
  fiveStarAuthors,
}) => (
  <main className="main-body">
    <section className="hero overlay">
      <div className="container">
        <div className="hero-inner">
          <h1>Write and share your ideas with the world</h1>
          <p>
              Join a community of like-minded authors to foster
              inspiration and innovation by leveraging the modern web.
          </p>
          <Link className="btn-brand" to="/signup">
              Get Started
          </Link>
        </div>
      </div>
    </section>

    <section>
      <div className="heading heading-no-spacing">
        <div className="container">
          <h3>
            <Link
                to="/categories"
                className="btn-brand btn-secondary transition"
              >
                More &gt;&gt;
            </Link>
              Browse by Category
          </h3>
        </div>
      </div>
      {loadingCategory === 'started' ? <PlaceholderLoader /> : <Category begin={0} end={6} articlesCategory={articlesCategory} />}

    </section>

    {Articles ? (
      <section>
        <div className="heading">
          <div className="container">
            <h3>
              <Link
                  to="/search"
                  className="btn-brand btn-secondary transition"
                >
                  More &gt;&gt;
              </Link>
                Recently Published
            </h3>
          </div>
        </div>
        {loadingArticles === 'started' ? <PlaceholderLoader /> : <Articles articles={allArticles} />}
      </section>
    ) : ''}

    {Authors ? (
      <section>
        <div className="heading">
          <div className="container">
            <h3>
              <Link
                  to="authors.html"
                  className="btn-brand btn-secondary transition"
                >
                  More &gt;&gt;
              </Link>
                Recommended Authors
            </h3>
          </div>
        </div>
        {loadingArticles === 'started' ? <PlaceholderLoader /> : Authors ? <Authors authors={fiveStarAuthors} begin={0} end={6} /> : ''}
      </section>
    ) : ''}


    <section className="cta">
      <div className="container">
        <div className="row">
          <div className="order-md-5 col-md-5">
            <div className="text-center">
              <img alt="" src={FooterImage} />
            </div>
          </div>
          <div className="col-md-7">
            <div className="cta-inner">
              <h2>Bring your creative writing skills to life</h2>
              <p>
                  With tools to make every part of your process more human
                  and a support team excited to help you, getting started
                  with inbound has never been easier.
              </p>
              <Link to="/signup" className="btn-brand">
                  Start Writing
              </Link>
            </div>
          </div>
        </div>
      </div>
    </section>
  </main>
);

Body.propTypes = {
  loadingCategory: Proptypes.string,
  PlaceholderLoader: Proptypes.func,
  Category: Proptypes.func,
  articlesCategory: Proptypes.array,
  loadingArticles: Proptypes.string,
  Articles: Proptypes.func,
  allArticles: Proptypes.array,
  Authors: Proptypes.func,
  fiveStarAuthors: Proptypes.array,
};

Body.defaultProps = {
  loadingCategory: '',
  PlaceholderLoader: f => f,
  Category: f => f,
  articlesCategory: [],
  loadingArticles: Proptypes.string,
  Articles: f => f,
  allArticles: Proptypes.array,
  Authors: f => f,
  fiveStarAuthors: [],
};

export default Body;
