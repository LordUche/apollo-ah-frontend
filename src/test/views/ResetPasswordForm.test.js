import React from 'react';
import { shallow } from 'enzyme';
import { ResetPasswordForm } from '../../views/ResetPasswordForm';

const props = {
  handleSubmit: jest.fn(),
  updateInput: jest.fn()
};

const event = {
  target: {
    name: 'password',
    value: 'mypassword'
  },
  preventDefault: jest.fn(),
};

describe('<ResetPasswordForm />', () => {
  it('should take a snapshot of the component', () => {
    const wrapper = shallow(<ResetPasswordForm {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle change event', () => {
    const wrapper = shallow(<ResetPasswordForm {...props} />);
    wrapper.find('.form-control').at(0).simulate('change', event);
    expect(props.updateInput).toHaveBeenCalled();
    wrapper.find('.form-control').at(1).simulate('change', event);
    expect(props.updateInput).toHaveBeenCalled();
  });
  it('should handle submit', () => {
    const wrapper = shallow(<ResetPasswordForm {...props} />);
    wrapper.find('.reset-password-form').at(0).simulate('submit', event);
    expect(props.handleSubmit).toHaveBeenCalled();
  });
});
