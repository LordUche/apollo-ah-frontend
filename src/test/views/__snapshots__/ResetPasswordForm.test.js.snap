// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPasswordForm /> should take a snapshot of the component 1`] = `
<div
  className="background min-vh-100"
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="login"
      >
        <div
          className="col-xs-12 col-sm-6 col-md-6 col-lg-6"
        >
          <div
            className="login__nav"
          >
            <Link
              href="/"
              replace={false}
              to="/"
            >
              ‚Üê
            </Link>
          </div>
          <div
            className="login__content"
          >
            <div
              className="login__intro"
            >
              <h3
                className="text-center"
              >
                Reset Password
              </h3>
              <p>
                Please fill in your new password to gain access to your account.
              </p>
            </div>
            <div
              className="login__input__section"
            >
              <form
                className="reset-password-form"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <div
                    className="login__input__field"
                  >
                    <label
                      htmlFor="email"
                    >
                      New Password
                    </label>
                    <input
                      className="form-control"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      type="password"
                    />
                  </div>
                  <div
                    className="login__input__field"
                  >
                    <div
                      className="login__forgot-password"
                    >
                      <label
                        htmlFor="password"
                      >
                        Confirm Password
                      </label>
                    </div>
                    <input
                      className="form-control"
                      id="confirm_password"
                      name="confirmPassword"
                      onChange={[Function]}
                      required={true}
                      type="password"
                    />
                  </div>
                  <div>
                    <button
                      className="form-control login__btn--large"
                      type="submit"
                    >
                      Reset Password
                    </button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          className="d-none d-sm-flex col-sm-6 col-md-6 col-lg-6 banner"
        >
          <div
            className="login__banner"
          >
            <div
              className="login__banner__content"
            >
              <hr />
              <div>
                <h3>
                  Hello,
                </h3>
                <p>
                  Log in to your account to start reading and writing beautiful articles.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
